soh/soh/Enhancements/randomizer/3drando/hints.cpp

```diff
@@ -695,15 +695,13 @@ std::vector<RandomizerCheck> FindItemsAndMarkHinted(std::vector<RandomizerGet> i
 
 void CreateChildAltarHint() {
   auto ctx = Rando::Context::GetInstance();
-  if (!ctx->GetHint(RH_ALTAR_CHILD)->IsEnabled()){
+  if (!ctx->GetHint(RH_ALTAR_CHILD)->IsEnabled() && ctx->GetOption(RSK_TOT_ALTAR_HINT)){
```

Comment:
This is because if RSK_TOT_ALTAR_HINT is false you'd end up with stoneLocs being empty, so stoneAreas would be empty, which would hit assertion error